### PR TITLE
Allow passing in a parameter to limit the number of transactions evaluated

### PR DESCRIPTION
**Pull request by jameinel**
This will let us do multiple passes for smaller transactions at a time,
rather than all at once.

https://bugs.launchpad.net/juju/+bug/1771906

### CODE REVIEW

File: oracle.go [hunk 1]
```diff
@@ -146,6 +153,7 @@ func (o *DBOracle) prepareWorkingDirectly() error {
 		docsToInsert = docsToInsert[:0]
 		return err
 	}
+	count := uint64(0)
```
Comment (manadart) [inline oracle.go]: We do not appear to be doing anything with this...

File: oracle_test.go [hunk 2]
```diff
@@ -169,6 +169,30 @@ func (s *OracleSuite) TestDoesntSeeNewTransactions(c *gc.C) {
...
+	const txnCount = 50
+	txnIds := make([]bson.ObjectId, txnCount)
+	for i := 0; i < txnCount; i++ {
+		txnIds[i] = s.runTxn(c, txn.Op{
+			C:      "coll",
+			Id:     i,
+			Insert: bson.M{},
+		})
+	}
+	// We should only find 10 of the 100 txns.
```
Comment (manadart) [inline oracle_test.go]: "... of the 50..."

File: prune.go [hunk 3]
```diff
@@ -50,7 +50,6 @@ const (
 	// we aren't removing.
 	maxIterCount = 5
 
-	// maxMemoryTokens caps our in-memory cache. When it is full, we will
```
Comment (manadart) [inline prune.go]: Accidental delete?

File: oracle.go [hunk 4]
```diff
@@ -106,6 +108,8 @@ type DBOracle struct {
 	txns            *mgo.Collection
 	working         *mgo.Collection
 	thresholdTime   time.Time
+	maxTxns         uint64
+	shouldRetry     bool
```
Comment (manadart) [inline oracle.go]: Can't see where this is used...

File: oracle.go (hunk 1 above)
Comment (jameinel) [reply oracle.go]: you're right. Original count was used to determine shouldRetry, but that was switched in favor of just checking Oracle.Count()

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**